src/arkode/arkode_erkstep.c

```diff
@@ -1273,6 +1566,295 @@ int erkStep_GetOrder(ARKodeMem ark_mem)
   return step_mem->q;
 }
 
+/*---------------------------------------------------------------
+  Utility routines for interfacing with SUNAdjointStepper
+  ---------------------------------------------------------------*/
+
+int erkStep_fe_Adj(sunrealtype t, N_Vector sens_partial_stage,
+                   N_Vector sens_complete_stage, void* content)
+{
+  SUNErrCode errcode = SUN_SUCCESS;
+
+  SUNAdjointStepper adj_stepper           = (SUNAdjointStepper)content;
+  SUNAdjointCheckpointScheme check_scheme = adj_stepper->checkpoint_scheme;
+  ARKodeMem ark_mem = (ARKodeMem)adj_stepper->adj_sunstepper->content;
+  void* user_data   = adj_stepper->user_data;
+
+  N_Vector Lambda_part     = N_VGetSubvector_ManyVector(sens_partial_stage, 0);
+  N_Vector Lambda          = N_VGetSubvector_ManyVector(sens_complete_stage, 0);
+  N_Vector checkpoint      = N_VGetSubvector_ManyVector(ark_mem->tempv3, 0);
+  sunrealtype checkpoint_t = SUN_RCONST(0.0);
+
+  errcode = SUNAdjointCheckpointScheme_LoadVector(check_scheme,
+                                                  ark_mem->adj_step_idx,
+                                                  ark_mem->adj_stage_idx, 0,
+                                                  &checkpoint, &checkpoint_t);
+
+  // Checkpoint was not found, recompute the missing step
+  if (errcode == SUN_ERR_CHECKPOINT_NOT_FOUND) { return +1; }
+
+  if (adj_stepper->JacFn)
+  {
+    adj_stepper->JacFn(t, checkpoint, NULL, adj_stepper->Jac, user_data, NULL,
+                       NULL, NULL);
+    adj_stepper->njeval++;
+    if (SUNMatHermitianTransposeVec(adj_stepper->Jac, Lambda_part, Lambda))
+    {
+      return -1;
+    };
+  }
+  else if (adj_stepper->JvpFn)
+  {
+    adj_stepper->JvpFn(Lambda_part, Lambda, t, checkpoint, NULL, user_data, NULL);
```

Comment:
Same `NULL` and temp vector comments as [here](https://github.com/LLNL/sundials/pull/649/files?show-viewed-files=true&file-filters%5B%5D=#r2019463113)